core/inc/oop_module.h

```diff
@@ -0,0 +1,38 @@
+
+#ifndef OOP_MODULE_H
+#define OOP_MODULE_H
+
+typedef OOP_MODULE * OOP_MODULE_HANDLE;
+
+extern
```

Comment:
I have added the appropriate decorations (i.e. MOCKABLE_FUNCTION and GATEWAY_EXPORT).